{{cookiecutter.project_slug}}/fabfile.py

```diff
@@ -1,144 +1,233 @@
+
+        will_write = force
+        if not force:
+            try:
+                # Test that it does exist
+                self.run_in_project_root("test -r {}".format(envfile_path), hide=True)
+            except UnexpectedExit:
+                will_write = True
+
+        if will_write:
+            if value is None:
+                value = input("Value for {}: ".format(name))
+
+            # Convert booleans into ints, so that Django takes them back easily
+            if isinstance(value, bool):
+                value = int(value)
+            self.put(StringIO("{}\n".format(value)), envfile_path)
+
+    def dump_db(self, destination):
+        """
+        Dump the database to the given directory and return the path to the file created.
+        This creates a gzipped SQL file.
+        """
+        with self.cd(self.project_root):
+            db_credentials = self.run(
+                "cat envdir/DATABASE_URL", hide=True
+            ).stdout.strip()
+        db_credentials_dict = dj_database_url.parse(db_credentials)
+
+        if not is_supported_db_engine(db_credentials_dict["ENGINE"]):
+            raise NotImplementedError(
+                "The dump_db task doesn't support the remote database engine"
+            )
 
-def collect_static():
-    """
-    Collect static files to the STATIC_ROOT directory.
-    """
-    with cd(get_project_root()):
-        run_python("manage.py collectstatic --noinput")
+        outfile = os.path.join(
+            destination, datetime.now().strftime("%Y-%m-%d_%H%M%S.sql.gz")
+        )
 
+        self.run(
+            "pg_dump -O -x -h {host} -U {user} {db}|gzip > {outfile}".format(
+                host=db_credentials_dict["HOST"],
+                user=db_credentials_dict["USER"],
+                db=db_credentials_dict["NAME"],
+                outfile=outfile,
+            ),
+            env={"PGPASSWORD": db_credentials_dict["PASSWORD"].replace("$", "\$")},
+        )
 
-def restart_process():
-    """
-    Restart the WSGI process
-    """
-    run("uwsgi --reload %s" % env.pid)
+        return outfile
+
+    def create_structure(self):
+        """
+        Create the basic directory structure on the remote server.
+        """
+        self.run("mkdir -p %s" % self.project_root)
+
+        with self.cd(self.site_root):
+            self.run("mkdir -p static backups media")
+
+    def clean_old_database_backups(self, nb_backups_to_keep):
+        """
+        Remove old database backups from the system and keep `nb_backups_to_keep`.
+        """
+        backups = self.ls(self.backups_root)
+        backups = sorted(backups, reverse=True)
+
+        if len(backups) > nb_backups_to_keep:
+            backups_to_delete = backups[nb_backups_to_keep:]
+            file_to_remove = [
+                os.path.join(self.backups_root, backup_to_delete)
+                for backup_to_delete in backups_to_delete
+            ]
+            self.run('rm "%s"' % '" "'.join(file_to_remove))
```

Comment:
Sorry to hijack the review, but I think it would be safer (and simpler in this case) to use `shlex.join`, i.e. `self.run('rm -- %s' % shlex.join(file_to_remove))`

Reply:
this feature is only implemented on python 3.8
https://docs.python.org/3/library/shlex.html#shlex.join